Source/WebCore/css/StyleProperties.cpp

```diff
@@ -336,12 +336,14 @@ String StyleProperties::getPropertyValue(CSSPropertyID propertyID) const
         return fontVariantValue();
     case CSSPropertyFontSynthesis:
         return fontSynthesisValue();
-    case CSSPropertyTextDecoration:
-        if (auto line = getPropertyCSSValue(CSSPropertyTextDecorationLine))
+    case CSSPropertyTextDecoration: {
+        auto line = getPropertyCSSValue(CSSPropertyTextDecorationLine);
+
+        if (is<CSSPrimitiveValue>(line) && downcast<CSSPrimitiveValue>(*line).valueID() == CSSValueNone)
```

Comment:
You can't just return `none` without checking that the other longhands are set to their initial value. E.g. `text-decoration: dotted` should not serialize as `none`.

Reply:
I'm not planning to finish this anytime soon fwiw. Feel free to take over

Reply:
I won't take over, handling the editing part without perf regressions seems tricky...

Reply:
@Loirooriol I suspect the regression was due to the parts in `void EditingStyle::removeEquivalentProperties(T& style)` being done for _all_ properties, including font, etc. I think my check to only do it for text-decoration should make things better.

Though I hadn't had time to look at the failures that happened after rebasing (and fixing the serialization).